mapclientplugins/geometricfitstep/view/geometricfitwidget.py

```diff
@@ -453,207 +482,287 @@ def _displaySurfacesTranslucentClicked(self):
-        """
-        Callback for change in model coordinates field chooser widget.
+        Update and display group setting widgets for currentFitterStep
         """
-        field = self._ui.configModelCoordinates_fieldChooser.getField()
-        if field:
-            self._fitter.setModelCoordinatesField(field)
+        #isAlign = isinstance(self._currentFitterStep, FitterStepAlign)
+        isConfig = isinstance(self._currentFitterStep, FitterStepConfig)
+        isFit = isinstance(self._currentFitterStep, FitterStepFit)
+        if isConfig:
+            self._updateCentralProjection()
+            self._updateDataProportion()
+        elif isFit:
+            self._updateDataWeight()
+            self._updateStrainPenalty()
+            self._updateCurvaturePenalty()
+        self._ui.configCentralProjection_checkBox.setVisible(isConfig)
+        self._ui.configSetCentralProjection_checkBox.setVisible(isConfig)
+        self._ui.configDataProportion_checkBox.setVisible(isConfig)
+        self._ui.configDataProportion_lineEdit.setVisible(isConfig)
+        self._ui.configDataWeight_checkBox.setVisible(isFit)
+        self._ui.configDataWeight_lineEdit.setVisible(isFit)
+        self._ui.configStrainPenalty_checkBox.setVisible(isFit)
+        self._ui.configStrainPenalty_lineEdit.setVisible(isFit)
+        self._ui.configCurvaturePenalty_checkBox.setVisible(isFit)
+        self._ui.configCurvaturePenalty_lineEdit.setVisible(isFit)
 
-    def _configDataCoordinatesFieldChanged(self, index):
+    def _configSettingGroupChanged(self, index):
         """
-        Callback for change in data coordinates field chooser widget.
+        Callback for change in setting group field chooser widget.
         """
-        field = self._ui.configDataCoordinates_fieldChooser.getField()
-        if field:
-            self._fitter.setDataCoordinatesField(field)
+        self._updateGroupSettingWidgets()
 
-    def _configMarkerGroupChanged(self, index):
-        """
-        Callback for change in marker group field chooser widget.
-        """
-        group = self._ui.configMarkerGroup_fieldChooser.getField()
+    def _getGroup(self):
+        group = self._ui.configSettingGroup_fieldChooser.getField()
         if group:
-            self._fitter.setMarkerGroup(group)
+            group = group.getName()
+        return group
 
-    def _configDiagnosticLevelValueChanged(self, value):
-        self._fitter.setDiagnosticLevel(value)
+    def _getGroupSettingDisplayState(self, func):
+        group = self._getGroup()
+        data, isLocallySet, inheritable = func(group)
+        realFormat = "{:.4g}"
+        lineEditDisable = True
+        checkBoxTristate = False
+        checkBoxState = QtCore.Qt.Unchecked
+        if isinstance(data, bool):
+            pass
+        elif isinstance(data, list):
+            data = ','.join(str(e) for e in data)
+        else:
+            data = realFormat.format(data)
```

Comment:
It would read better to check and process the actual types, with the bool case handled with the final else:
```
else:
    assert isinstance(data, bool)
```
This means any other types we add will not silently fail.